src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -14,89 +14,86 @@ namespace Microsoft.Authentication.AzureAuth
-            {
-                if (exception.GetType() == typeof(MsalServiceException))
-                {
-                    var msalServiceException = (MsalServiceException)exception;
-                    if (msalServiceException.CorrelationId != null)
-                    {
-                        correlationIDs.Add(msalServiceException.CorrelationId);
-                    }
-                }
-                else if (exception.GetType() == typeof(MsalUiRequiredException))
-                {
-                    var msalUiRequiredException = (MsalUiRequiredException)exception;
-                    if (msalUiRequiredException.CorrelationId != null)
-                    {
-                        correlationIDs.Add(msalUiRequiredException.CorrelationId);
-                    }
-                }
-            }
-
-            return correlationIDs;
-        }
-
         /// <summary>
         /// Executes the convertion of exceptions to a JSON string.
         /// </summary>
         /// <param name="exceptions">List of exceptions.</param>
         /// <returns>JSON format of exception list.</returns>
         public static string Execute(IEnumerable<Exception> exceptions)
         {
-            if (exceptions == null || exceptions.Count() == 0)
-            {
-                return null;
-            }
-
-            List<SerializableException> customExceptions = new List<SerializableException>();
+            List<SerializableException> serializableExceptions = new List<SerializableException>();
             foreach (Exception exception in exceptions)
             {
                 var customException = new SerializableException(exception);
-                customExceptions.Add(customException);
+                serializableExceptions.Add(customException);
             }
 
-            return JsonSerializer.Serialize(customExceptions);
+            return JsonSerializer.Serialize(serializableExceptions);
         }
-    }
 
-    /// <summary>
-    /// CustomException class with only select properties of Exception class.
-    /// </summary>
-    internal class SerializableException
-    {
         /// <summary>
-        /// Initializes a new instance of the <see cref="SerializableException"/> class.
+        /// CustomException class with only select properties of Exception class.
         /// </summary>
-        /// <param name="exception">Exception.</param>
-        public SerializableException(Exception exception)
+        public class SerializableException
         {
-            if (exception != null)
+            /// <summary>
+            /// Initializes a new instance of the <see cref="SerializableException"/> class.
+            /// </summary>
+            /// <param name="exception">Exception.</param>
+            public SerializableException(Exception exception)
             {
+                if (exception == null)
+                {
+                    return;
+                }

```

Comment:
Does this need to return null, or is handing this a null value an invalid state? Can we throw instead?

Reply:
Yea, since we are calling recursively on inner exceptions we need this.

Reply:
Or may be that check can be done before the recursive call

Reply:
@kyle-rader I have addressed the new comments :)